documentation for HTMLMapElement.areas (#31713)
mdn/content: files/en-us/web/api/htmlmapelement/areas/index.md

---
title: "HTMLMapElement: areas property"
short-title: areas
slug: Web/API/HTMLMapElement/areas
page-type: web-api-instance-property
browser-compat: api.HTMLMapElement.areas
---

{{ApiRef("HTML DOM")}}

> The **`areas`** read-only property of the {{domxref("HTMLMapElement")}} return {{domxref("HTMLCollection")}} object of {{HTMLElement("area")}} elements present in the {{HTMLElement("map")}} element. This property only apply filter for `area` elements. We can not assign a new collection to `areas` property, we can only access or iterate over existing areas collections.

Reviewer: Read-only is clear enough: no need to explain you can't modify it.

```suggestion
The **`areas`** read-only property of the {{domxref("HTMLMapElement")}} elements returns a collection of {{HTMLElement("area")}} elements associated with the {{HTMLElement("map")}} element. 

Two successive calls to this property will return the same object.
```

## Value

> An array-like {{domxref("HTMLCollection")}} object of `area` elements.

Reviewer: ```suggestion
A live {{domxref("HTMLCollection")}} of {{domxref("HTMLAreaElement")}} objects.
```

Author: ```suggestion
A  present {{domxref("HTMLCollection")}} of {{HTMLElement("area")}} objects.
```
`live` is not a proper word, what about `present' word?

> An array-like object ({{domxref("HTMLCollection")}}) of elements.

Author: it is useful for beginners

> An array-like object ({{domxref("HTMLCollection")}}) of elements.

Reviewer 2: But beside this, I don't think HTMLCollection is array-like

> An array-like object ({{domxref("HTMLCollection")}}) of elements.

Author: This word is from mdn [HTMLCollection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection). 
Htmlcollections are array like object similar to 'arguments'. 

I thought if it is helpful for readers to follow content from other sources on MDN. 

In my opinion, Generic collection or just HTMLCollection are the most correct words. What is your opinion?

> An array-like object ({{domxref("HTMLCollection")}}) of elements.

Reviewer 2: Well, it seems ok for me now

> An array-like object ({{domxref("HTMLCollection")}}) of elements.

Author: Ok, I will wait for @teoli2003 suggestions.

## Example

```html
<map name="image-map">
  <a href="www.link1_example.com">link 1</a>
  <area
    alt="left arrow"
    shape="circle"
    coords="50,50,35"
    href="www.left_example.com"
    target="_top" />
  <a href="www.link2_example.com">link 2</a>
  <area
    alt="right arrow"
    shape="circle"
    coords="150,50,35"
    href="www.right_example.com"
    target="_top" />
</map>
<img
  usemap="#image-map"
  src="left-right-arrow.png"
  alt="left right arrow image" />
```

```js
const mapElement = document.getElementsByName("image-map")[0];
// Anchor elements are filtered out from HTMLCollection

> console.log(mapElement.areas); // output: HTMLCollection(2) [area, area]

Reviewer: ```suggestion
console.log(mapElement.areas); // Output: HTMLCollection(2) [area, area]
```

const mapElementAreas = mapElement.areas;

> // iterate over each area element and change target value to _blank

Reviewer: ```suggestion
// Iterate over each area element and change target value to _blank
```

[...mapElementAreas].forEach((area) => (area.target = "_blank"));
console.log(mapElementAreas[0]);

> // click on left and right arrows, it will open a new tab.

Reviewer: As a live example, I don't think to make them open in a new tab is good.

Can we find something better here?

Author: Then I will make dynamic changes in 'alt'.
